tools/health/health_checker_pool.go

```diff
@@ -247,3 +270,97 @@ func (ncp *NodeHealthCheckerPool) WaitAllHashesEqual(height uint64) error {
 		}
 	}
 }
+
+func (ncp *NodeHealthCheckerPool) FindInconsistentHashesAtHeight(height uint64) map[string]sdk.Hash {
```

Comment:
I think, you can just slightly improve the `CheckHashes` method. It already checks hashes and return just bool. You can make it return a bool and map of hashes. If there is one hash - all is OK, more - there is a fork